asyncscope.md

```diff
@@ -2390,40 +2390,126 @@ creates an asynchronous operation (__[async.ops]__) that, when started:
-`decltype((sndr))`, let `Token` be `decltype((token))`, and let `Env` be `decltype((env))`. If `sender<Sndr>` or `async_scope_token<Snder>` is false, 
-the expression `spawn(sndr, token, env)` is ill-formed. 
+[2]{.pnum} The name `spawn` denotes a customization point object. For subexpressions `sndr`, `token`, and `env`, let `Sndr` be
+`decltype((sndr))`, let `Token` be `decltype((token))`, and let `Env` be `decltype((env))`. If `sender<Sndr>` or `async_scope_token<Token>` is false,
+the expression `spawn(sndr, token, env)` is ill-formed.
 
-// TODO: when/how do I specify that env is optional?  Or do we get the env from sender-in?
+[3]{.pnum} For the expression `spawn(sndr, token, env)` let _`new-sender`_ be the expression `token.wrap(sndr)` and let `alloc` and `senv` be defined as follows:
+- if the expression `get_allocator(env)` is well defined, then `alloc` is the result of `get_allocator(env)` and `senv` is the expression `env`,
+- otherwise if the expression `get_allocator(get_env(@_new-sender_@))` is well-defined, then `alloc` is the result of `get_allocator(get_env(@_new-sender_@))`
+  and `senv` is the expression `@_JOIN-ENV_@(env, @_MAKE-ENV_@(get_allocator, alloc))`
+- otherwise `alloc` is `std::allocator<void>{}` and `senv` is the expression `env`
 
-`[3]{.pnum} The exposition-only class template `impls-for` ([exec.snd.general])  is specialized for `spawn_t` as follows:
+[4]{.pnum} Let _`spawn-state-base`_ be an expositon only class defined below:
 
 ```cpp
 namespace std::execution {
-template<>
-struct impls-for<spawn_t> : default_impls {
-  static constexpr auto get-env = see below;
-  static constexpr auto start = see below;
-  static constexpr auto complete = see below;
+struct @_spawn-state-base_@ { // exposition-only
+    virtual void complete() = 0; // exposition-only
 };
 }
 ```
 
-- [3.1]{.pnum} The member `impls-for<spawn_t>::get-env` is initialized with
-// TODO: how to express what get-env does. Also is this the right exposition-only function?
+[5]{.pnum} Let _`spawn-receiver`_ be an exposition only class defined below:
+```cpp
+namespace std::execution {
+struct @_spawn-receiver_@ { // exposition-only
+    @_spawn-state-base_@* state; // exposition-only
+    void set_value() && noexcept { state->complete(); }
+    void set_stopped() && noexcept { state->complete(); }
+};
+}
+```
+
+[6]{.pnum} Let_`spawn-state`_ be an exposition only class template defined
+below:
+
+```cpp
+namespace std::execution {
+template<class Alloc, async_scope_token Token, sender Sender>
+struct @_spawn-state_@ : @_spawn_state_base_@ {
+    using Op = decltype(connect(declval<Sender>(), spawn-receiver{nullptr}));
 
-- [3.2]{.pnum} The member `impls-for<spawn_t>::start` is initialized with a callable object equivalent to the following lambda expression:
-// TODO: is this the right exposition only function?
+    @_spawn-state_@(Alloc alloc, Sender sndr, Token token); // see below
```

Comment:
Section 6 - 9 are inspired by split in P2300: https://eel.is/c++draft/exec#split-5

I'm leaving in section 10 for now in case section 6-9 needs to be redone.